SignalServiceKit/src/Contacts/TSThread.m

```diff
@@ -395,16 +412,58 @@ - (BOOL)isMuted
             [mutedUntilDate timeIntervalSinceDate:now] > 0);
 }
 
-#pragma mark - Update With... Methods
+- (void)updateWithMutedUntilDate:(NSDate *)mutedUntilDate transaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    [self applyChangeToSelfAndLatestCopy:transaction
+                             changeBlock:^(TSThread *thread) {
+                                 [thread setMutedUntilDate:mutedUntilDate];
+                             }];
+}
+
+#pragma mark - Conversation Color
 
-- (void)updateWithMutedUntilDate:(NSDate *)mutedUntilDate
++ (NSString *)randomConversationColorName
 {
-    [self.dbReadWriteConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
-        [self applyChangeToSelfAndLatestCopy:transaction
-                                 changeBlock:^(TSThread *thread) {
-                                     [thread setMutedUntilDate:mutedUntilDate];
-                                 }];
-    }];
+    NSUInteger count = self.conversationColorNames.count;
+    NSUInteger index = arc4random_uniform((uint32_t)count);
+    return [self.conversationColorNames objectAtIndex:index];
+}
+
++ (NSString *)stableConversationColorNameForString:(NSString *)colorSeed
+{
+    NSData *contactData = [colorSeed dataUsingEncoding:NSUTF8StringEncoding];
+    
+    NSUInteger hashingLength = 8;
+    unsigned long long choose;
+    NSData *hashData = [Cryptography computeSHA256Digest:contactData truncatedToBytes:hashingLength];
+    [hashData getBytes:&choose length:hashingLength];
+    
+    NSUInteger index = (choose % [self.conversationColorNames count]);
+    return [self.conversationColorNames objectAtIndex:index];
+}
+
++ (NSArray<NSString *> *)conversationColorNames
+{
+    return @[
```

Comment:
I felt bad just copying these in, but I wasn't sure what the best way to dedupe was.

We could an assert that does some accounting, we could pull UIColor+OWS into SSK, or we could have a separate class that lives in SSK which UIColor+OWS uses as a helper...